### ISSUE TITLE
[BUG] Failed to get updates in python 3.9

### ISSUE THREAD

**Issue by josephchenhk** [python-telegram-bot/python-telegram-bot#2838]
### Steps to Reproduce

1. `import telegram

bot = telegram.Bot(token="TOKEN")
print(bot.get_me())

updates = bot.get_updates()
print(updates[0])`

2. This line `updates = bot.get_updates()` fails, with following error: TypeError: changelist must be an iterable of select.kevent objects

### Expected behaviour

Expect to work as the documentation: https://github.com/python-telegram-bot/python-telegram-bot/wiki/Introduction-to-the-API

### Actual behaviour

This line `updates = bot.get_updates()` fails, with following error: TypeError: changelist must be an iterable of select.kevent objects

### Operating System

Mac OS

### Version of Python, python-telegram-bot & dependencies

```shell
Python==3.9, python-telegram-bot==13.9
```

**Comment by harshil21** [comment]
Hi,

Can you post the entire log output? Does not seem related to our library at first glance.

**Comment by josephchenhk** [comment]
> Hi,
> 
> Can you post the entire log output? Does not seem related to our library at first glance.

Yes,  the full log is as below:

Traceback (most recent call last):
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/utils/request.py", line 259, in _request_wrapper
    resp = self._con_pool.request(*args, **kwargs)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/request.py", line 68, in request
    return self.request_encode_body(method, url, fields=fields,
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/request.py", line 148, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/poolmanager.py", line 244, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/connectionpool.py", line 605, in urlopen
    conn = self._get_conn(timeout=pool_timeout)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/connectionpool.py", line 242, in _get_conn
    if conn and is_connection_dropped(conn):
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/connection.py", line 27, in is_connection_dropped
    return bool(wait_for_read(sock, timeout=0.0))
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/wait.py", line 33, in wait_for_read
    return _wait_for_io_events(socks, EVENT_READ, timeout)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/wait.py", line 24, in _wait_for_io_events
    selector.register(sock, events)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/selectors.py", line 448, in register
    _syscall_wrapper(self._kqueue.control, False, [kevent], 0, 0)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/selectors.py", line 81, in _syscall_wrapper
    result = func(*args, **kwargs)
TypeError: changelist must be an iterable of select.kevent objects

**Comment by josephchenhk** [comment]
> Hi,
> 
> Can you post the entire log output? Does not seem related to our library at first glance.

Yes, the full log is as below:

`
Traceback (most recent call last):
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/utils/request.py", line 259, in _request_wrapper
    resp = self._con_pool.request(*args, **kwargs)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/request.py", line 68, in request
    return self.request_encode_body(method, url, fields=fields,
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/request.py", line 148, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/poolmanager.py", line 244, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/connectionpool.py", line 605, in urlopen
    conn = self._get_conn(timeout=pool_timeout)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/connectionpool.py", line 242, in _get_conn
    if conn and is_connection_dropped(conn):
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/connection.py", line 27, in is_connection_dropped
    return bool(wait_for_read(sock, timeout=0.0))
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/wait.py", line 33, in wait_for_read
    return _wait_for_io_events(socks, EVENT_READ, timeout)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/wait.py", line 24, in _wait_for_io_events
    selector.register(sock, events)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/selectors.py", line 448, in register
    _syscall_wrapper(self._kqueue.control, False, [kevent], 0, 0)
  File "/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/selectors.py", line 81, in _syscall_wrapper
    result = func(*args, **kwargs)
TypeError: changelist must be an iterable of select.kevent objects
`

**Comment by Bibo-Joshi** [comment]
does that happen every time you call `get_updates` or does it happen only occasionally? did you install PTB from pypi (e.g. via pip) or did you install it from some conda repo?

**Comment by josephchenhk** [comment]
> does that happen every time you call `get_updates` or does it happen only occasionally? did you install PTB from pypi (e.g. via pip) or did you install it from some conda repo?

It happens every time I call `get_updates`. I didn't install PTB. It is a newly created conda environment python 3.9.

**Comment by Bibo-Joshi** [comment]
okay. can you try installing PTB manually from pypi? does that work on conda?

**Comment by josephchenhk** [comment]
> okay. can you try installing PTB manually from pypi? does that work on conda?

I installed ptb==0.0.3 in the environment now. The issue still exists.

**Comment by Bibo-Joshi** [comment]
uhm, we're on v13.10 by now. make sure to uninstall any currently existing version before installing v13.10. you can also install from source as described in the readme

**Comment by josephchenhk** [comment]
> uhm, we're on v13.10 by now. make sure to uninstall any currently existing version before installing v13.10. you can also install from source as described in the readme

Thanks. I tried installing v13.10 from source, the issue still exists.

**Comment by Bibo-Joshi** [comment]
Okay. This is indeed strange, since our test suite [passes on py 3.9 on macOS](https://github.com/python-telegram-bot/python-telegram-bot/runs/4688759137?check_suite_focus=true). BTW, does this happen only for `get_updates` or also for other bot methods, e.g. `send_message`?

---

Anyway, the traceback leads to this line

https://github.com/python-telegram-bot/urllib3/blob/1954df03958b164483282330b3a58092c070bc7a/urllib3/util/selectors.py#L81

 were `func`, `args` and `kwargs` are set here:

https://github.com/python-telegram-bot/urllib3/blob/1954df03958b164483282330b3a58092c070bc7a/urllib3/util/selectors.py#L443-L448

As you can see, `self._kqueue` is set to `select.kqueue()` in `__init__`, which is [this function](https://docs.python.org/3/library/select.html#select.kqueue) from the std lib. So the `TypeError: changelist must be an iterable of select.kevent objects` is raised by [`kqueue.control`](https://docs.python.org/3/library/select.html#select.kqueue.control) and the error message makes sense from that functions docstring. Now, the `changelist` argument as apparently set to `[kevent]`, where `kevent` is the return value of [`select.kevent`](https://docs.python.org/3/library/select.html#select.kevent) (line 444) - which is again a function from the std-lib.

From my understanding, the problem is that `kevent` is not the type of object that `kqueue.control` expects for some reason. If this is indeed the case, then this seems to be a problem of the std-libs `select` module in py3.9.

## TL;DR:
Please try to insert a `print(kevent, type(kevent)` before the `_syscall_wrapper` call in line 448 of the file `"/Users/joseph/miniconda3/envs/py39/lib/python3.9/site-packages/telegram/vendor/ptb_urllib3/urllib3/util/selectors.py"` and report the output of that.

**Comment by Bibo-Joshi** [comment]
Closing due to inactivity. Feel free to reply in order to repoen. This thread will be automatically locked in 7 days.
